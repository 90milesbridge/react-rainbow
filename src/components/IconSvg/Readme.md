IconSvg examples:
    
    <div>
        <div className="slds-p-bottom--medium">
            <IconSvg iconName="custom:custom5" wrapper="icon" style={{  backgroundColor: 'aqua' }} />
        </div>
        <div className="slds-p-bottom--medium">
            <IconSvg iconName="utility:like" wrapper="button" />
        </div>
        <div className="slds-p-bottom--medium">
            <IconSvg iconName="utility:warning" title="like button" variant="error" size="large" />
        </div>
    </div>