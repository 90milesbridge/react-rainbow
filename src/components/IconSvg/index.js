import React from 'react';
import PropTypes from 'prop-types';
import {
    isIconNameValid,
    getSpriteName,
    getIconName,
} from './utils';

export default function IconSvg(props) {
    const {
        className,
        style,
        iconName,
    } = props;

    if (isIconNameValid(iconName)) {
        const sprite = getSpriteName(iconName);
        const icon = getIconName(iconName);

        return (
            <svg className={className} style={style} aria-hidden>
                <use xmlnsXlink="http://www.w3.org/1999/xlink" xlinkHref={`/icons/${sprite}-sprite/svg/symbols.svg#${icon}`} />
            </svg>
        );
    }
    return null;
}

IconSvg.propTypes = {
    /** The Lightning Design System name of the icon used as a fallback when
    * the image fails to load. Names are written in the format {sprite_name}:{icon_name}
    * where {sprite_name} is the category, and {icon_name} is the specific icon to be displayed. */
    iconName: PropTypes.string.isRequired,
    /** A CSS class for the outer element, in addition to the component's base classes. */
    className: PropTypes.string,
    /** An object with custom style applied to the outer element. */
    style: PropTypes.object,
};

IconSvg.defaultProps = {
    className: '',
    style: {},
};
