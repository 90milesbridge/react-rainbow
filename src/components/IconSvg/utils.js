const FORMAT_REGEX = /^[a-z]+:[a-z_0-9]+$/;

export function isIconNameValid(iconName) {
    return FORMAT_REGEX.test(iconName);
}

export function getSpriteName(iconName) {
    return iconName.split(':')[0];
}

export function getIconName(iconName) {
    return iconName.split(':')[1];
}
