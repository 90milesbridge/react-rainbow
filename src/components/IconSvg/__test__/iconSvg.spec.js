import React from 'react';
import { mount } from 'enzyme';
import IconSvg from './../index';

describe('<IconSvg/>', () => {
    it('should return nothing when the iconName format is wrong', () => {
        const component = mount(
            <IconSvg iconName="wrong-name" />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should render the svg element when the iconName format is valid', () => {
        const component = mount(
            <IconSvg iconName="utility:like" />,
        );
        expect(component.find('svg').exists()).toBe(true);
    });
    it('should set the aria-hidden to true in svg element', () => {
        const component = mount(
            <IconSvg iconName="utility:like" />,
        );
        expect(component.find('svg').prop('aria-hidden')).toBe(true);
    });
    it('should set the right xlinkHref value in use element', () => {
        const component = mount(
            <IconSvg iconName="action:add_contact" />,
        );
        expect(component.find('use').prop('xlinkHref')).toBe('/icons/action-sprite/svg/symbols.svg#add_contact');
    });
});
