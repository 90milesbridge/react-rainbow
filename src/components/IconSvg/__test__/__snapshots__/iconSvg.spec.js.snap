// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconSvg/> should render a icon 1`] = `
<svg
  aria-hidden="true"
  className="slds-icon-text-default"
  style={undefined}
>
  <use
    xlinkHref="/icons/utility-sprite/svg/symbols.svg#like"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;

exports[`<IconSvg/> should render nothing when the iconName's format is wrong 1`] = `null`;
